test/java/org/chromium/net/CertificateVerificationTest.java

```diff
@@ -143,4 +145,24 @@ public void testChainWithMultipleRootCertificates() throws Exception {
                                                                                   authType, host);
     assertEquals(result.getStatus(), CertVerifyStatusAndroid.OK);
   }
+
+  @Test
```

Comment:
Nice tests! 

It's still unclear to me how the previous tests didn't run into any issue with the incorrect jobject being passed in. Can you explain a bit?

Reply:
I was also confused, but if you look closely at the tests above they don't exercise the path where C++ unpacks AndroidCertVerifyResult through JNI. They just construct the AndroidCertVerifyResult through JNI (`JniLibrary.callCertificateVerificationFromNative`) and then unpack that directly in Java, skipping the wrong JNI code...

The new tests generate the AndroidCertVerifyResult directly in Java, then unpack it through JNI calls, so now we should be covering the entirety of the JNI call surface.

Reply:
Wow, that's tricky. But nice that we found it out!